commons/src/test/java/io/aiven/kafka/tiered/storage/commons/storage/filesystem/FileSystemStorageFactoryTest.java

```diff
@@ -0,0 +1,103 @@
+import org.junit.jupiter.params.provider.ValueSource;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatNoException;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+class FileSystemStorageFactoryTest {
+    @ParameterizedTest
+    @NullSource
+    @ValueSource(booleans = { false })
+    void testNotAllowOverwriting(final Boolean overwriteEnabled,
+                                 @TempDir final Path tempDir) throws IOException {
+        final File fsRoot = tempDir.toFile();
+        final ObjectStorageFactory osFactory = new FileSystemStorageFactory();
+        final HashMap<String, String> config = new HashMap<>();
+        config.put("root", fsRoot.toString());
+        if (overwriteEnabled != null) {
+            config.put("overwrite.enabled", Boolean.toString(overwriteEnabled));
+        }
+        osFactory.configure(config);
+
+        final InputStream logFile = new ByteArrayInputStream("log file".getBytes());
+        osFactory.fileUploader().upload(logFile, "aaa/0.log.txt");
+
+        assertThatThrownBy(() -> osFactory.fileUploader().upload(logFile, "aaa/0.log.txt"))
+            .isInstanceOf(IOException.class)
+            .hasMessage("File %s already exists", fsRoot + "/aaa/0.log.txt");
+    }
+
+    @Test
+    void testAllowOverwriting(@TempDir final Path tempDir) throws IOException {
+        final File fsRoot = tempDir.toFile();
+        final ObjectStorageFactory osFactory = new FileSystemStorageFactory();
+        osFactory.configure(Map.of(
+            "root", fsRoot.toString(),
+            "overwrite.enabled", "true"
+        ));
+
+        final InputStream logFile = new ByteArrayInputStream("log file".getBytes());
+        osFactory.fileUploader().upload(logFile, "aaa/0.log.txt");
+        assertThatNoException().isThrownBy(() -> osFactory.fileUploader().upload(logFile, "aaa/0.log.txt"));
+    }
+
+    @Test
+    void testUploadFetchDelete(@TempDir final Path tempDir) throws IOException {
+        final File fsRoot = tempDir.toFile();
+        final ObjectStorageFactory osFactory = new FileSystemStorageFactory();
+        osFactory.configure(Map.of(
+            "root", fsRoot.toString()
+        ));
+
+        final byte[] data = "some file".getBytes();
+        final InputStream file = new ByteArrayInputStream(data);
+        osFactory.fileUploader().upload(file, "aaa/0.log.txt");
+
+        try (final InputStream fetch = osFactory.fileFetcher().fetch("aaa/0.log.txt")) {
+            final String r = new String(fetch.readAllBytes());
+            assertThat(r).isEqualTo("some file");
+        }
+
+        try (final InputStream fetch = osFactory.fileFetcher().fetch("aaa/0.log.txt", 1, data.length - 2)) {
+            final String r = new String(fetch.readAllBytes());
+            assertThat(r).isEqualTo("ome fil");
+        }
+
+        osFactory.fileDeleter().delete("aaa/0.log.txt");
```

Comment:
Perhaps return the deletion result and assert it here?

Reply:
`Files.delete` inside doesn't return, only throws on exception.